src/pybind/mgr/orchestrator/module.py

```diff
@@ -70,14 +70,26 @@ def to_yaml_n(objs):
         raise OrchestratorError(f'unsupported format type: {format}')
 
 
-def generate_preview_tables(data):
+def generate_preview_tables(data, osd_only=False):
     error = [x.get('error') for x in data if x.get('error')]
     if error:
         return json.dumps(error)
     warning = [x.get('warning') for x in data if x.get('warning')]
     osd_table = preview_table_osd(data)
     service_table = preview_table_services(data)
```

Comment:
Seems sensible do not calculate service_table if you are not going to use it. What about to move the creation of the preview tables inside the right part of the conditional?